### PR TITLE
[DOC-8971] Update docs for renamed shutdown settings

### LINKED ISSUE #18684: Feedback: What's New in v23.2
Page: https://cockroachlabs.com/docs/releases/v23.2.html

## What is the reason for your feedback?

[ ] Missing the information I need

[ ] Too complicated

[ ] Out of date

[ ] Something is broken

[x] Other

## Additional details
The connection.timeout and transaction_timeout alias aren’t correct, comparing w/ cluster setting page.
Can someone review and correct it please?

### PR DESCRIPTION
**Pull request by mdlinville**
Fixes #18684
Fixes [DOC-8971]

Now that Cluster Settings pages show both the old and new names and the deprecation message has been softened, I can update the docs to use the new names.

- Also correct setting name errors in old release notes
- Also add details about key shutdown settings to the Cloud Production Checklist, per Carlo

Previews:
[releases/v23.2.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/releases/v23.2.html)
[src/current/v23.2/monitoring-and-alerting.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v23.2/monitoring-and-alerting.html)
[src/current/v23.2/node-shutdown.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v23.2/node-shutdown.html)
[src/current/v24.1/monitoring-and-alerting.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.1/monitoring-and-alerting.html)
[src/current/v24.1/node-shutdown.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.1/node-shutdown.html)
[src/current/v24.2/monitoring-and-alerting.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.2/monitoring-and-alerting.html)
[src/current/v24.2/node-shutdown.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.2/node-shutdown.html)
[src/current/cockroachcloud/production-checklist.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/cockroachcloud/production-checklist.html)

### CODE REVIEW

File: src/current/cockroachcloud/production-checklist.md [hunk 1]
```diff
@@ -66,6 +66,20 @@ When several transactions try to modify the same underlying data concurrently, t
 
 To ensure optimal SQL performance for your CockroachDB {{ site.data.products.cloud }} cluster, follow the best practices described in the [SQL Performance Best Practices](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/performance-best-practices-overview) guide.
 
+## Configure node shutdown settings
+
+Clusters in CockroachDB {{ site.data.products.cloud }} have the following default values for [cluster settings](../stable/cluster-settings) that relate to [node shutdown](../stable/node-shutdown) for maintenance, upgrades, or scaling. Depending on the requirements of your applications and workloads, you may need to modify `server.shutdown.connections.timeout` or `server.shutdown.transactions.timeout`.
```
Comment (mdlinville) [inline src/current/cockroachcloud/production-checklist.md:71]: @carloruiz PTAL

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md [hunk 2]
```diff
@@ -105,8 +105,8 @@ Release Date: October 10, 2023
     |---------------------------------------|----------------------------------------------------|
     | `server.shutdown.drain_wait`          | `server.shutdown.initial_wait`                     |
     | `server.shutdown.lease_transfer_wait` | `server.shutdown.lease_transfer_iteration.timeout` |
-    | `server.shutdown.query_wait`          | `server.shutdown.queries.timeout`                  |
-    | `server.shutdown.connection_wait`     | `server.shutdown.transactions.timeout`             |
+    | `server.shutdown.query_wait`          | `server.shutdown.transactions.timeout`             |
```
Comment (rmloveland) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md:108]: I do not understand this. Were the previous release notes incorrect?  

If we go back and say this about the v23.2.0-alpha.3 either the docs were wrong (totally possible) or ... ?

File: src/current/cockroachcloud/production-checklist.md [hunk 3]
```diff
@@ -66,6 +66,20 @@ When several transactions try to modify the same underlying data concurrently, t
 
 To ensure optimal SQL performance for your CockroachDB {{ site.data.products.cloud }} cluster, follow the best practices described in the [SQL Performance Best Practices](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/performance-best-practices-overview) guide.
 
+## Configure node shutdown settings
```
Comment (rmloveland) [inline src/current/cockroachcloud/production-checklist.md:69]: this section LGTM from a docs editing POV but I defer to @carloruiz for technical review

**Review by rmloveland — APPROVED** [review]
LGTM from Docs POV modulo any technical review comments that may need addressing 

EDIT: oh and I need to understand why we are back changing those old alpha release notes, that doesn't make sense to me unless those docs were actually wrong

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md (hunk 2 above)
Comment (mdlinville) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md:108]: The release note was incorrect. This fixes it.

File: src/current/cockroachcloud/production-checklist.md [hunk 4]
```diff
@@ -66,6 +66,20 @@ When several transactions try to modify the same underlying data concurrently, t
...
+
+Clusters in CockroachDB {{ site.data.products.cloud }} have the following default values for [cluster settings]({% link {{ site.current_cloud_version }}/cluster-settings.md %}) that relate to [node shutdown]({% link {{ site.current_cloud_version }}/node-shutdown.md %}) for maintenance, upgrades, or scaling. Depending on the requirements of your applications and workloads, you may need to modify `server.shutdown.connections.timeout` or `server.shutdown.transactions.timeout`.
+
+Cluster setting | Default | Details
+----------------|---------|---------
+[`server.shutdown.initial_wait`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-drain-wait)<br />Alias: `server.shutdown.drain_wait` | 3 minutes | How long to wait before redirecting new client requests to non-draining nodes. After this duration has elapsed, the node no longer accepts new client connections. This setting cannot be changed for clusters in CockroachDB {{ site.data.products.cloud }}.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:75]: I'd prefer not to mention this as it is not actionable for the customer and makes the doc harder to digest.

File: src/current/cockroachcloud/production-checklist.md [hunk 5]
```diff
@@ -66,6 +66,20 @@ When several transactions try to modify the same underlying data concurrently, t
...
+[`server.shutdown.transactions.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-query-wait)<br />Alias: `server.shutdown.query_wait` | 90 seconds | How long to wait for transactions on the node to complete before rolling them back. The default value is appropriate for most deployments. If this is set too low, transactions will be rolled back unnecessarily during node shutdown, and if it is set too high, node shutdown may take longer.
+
+If node termination takes longer than 45 minutes, regardless of the cluster's settings, the node is forcibly terminated and restarted.
+
+After a node is terminated, if it takes longer than 5 minutes to rejoin the cluster (known as the _termination grace period_), its replicas will be moved to other nodes and it will join as a new node.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:81]: I'd also not mention this as it's not actionable for customers and sounds scary.

File: src/current/cockroachcloud/production-checklist.md (hunk 3 above)
Comment (carloruiz) [reply src/current/cockroachcloud/production-checklist.md:69]: I'd move this section below `Keeping connection current` since the desired value for `connection_wait` should be set equal to or above the max lifetimes set on their connections.

File: src/current/cockroachcloud/production-checklist.md [hunk 6]
```diff
@@ -66,6 +66,20 @@ When several transactions try to modify the same underlying data concurrently, t
...
+[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds | How long after `server.shutdown.initial_wait` completes to wait for client connections to drain before forcibly disconnecting them from the node. Can be set to a value between 0 and 30 minutes (1800 seconds).
+[`server.shutdown.transactions.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-query-wait)<br />Alias: `server.shutdown.query_wait` | 90 seconds | How long to wait for transactions on the node to complete before rolling them back. The default value is appropriate for most deployments. If this is set too low, transactions will be rolled back unnecessarily during node shutdown, and if it is set too high, node shutdown may take longer.
+
+If node termination takes longer than 45 minutes, regardless of the cluster's settings, the node is forcibly terminated and restarted.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:79]: The hard limit for termination should not be communicated externally as we (the infra team) want to be flexible with this value. We should distill the termination info through the cluster settings only to avoid unnecessary complexity on the customer side and avoid customer asks that we can't accommodate.

File: src/current/cockroachcloud/production-checklist.md [hunk 7]
```diff
@@ -66,6 +66,20 @@ When several transactions try to modify the same underlying data concurrently, t
...
+
+Cluster setting | Default | Details
+----------------|---------|---------
+[`server.shutdown.initial_wait`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-drain-wait)<br />Alias: `server.shutdown.drain_wait` | 3 minutes | How long to wait before redirecting new client requests to non-draining nodes. After this duration has elapsed, the node no longer accepts new client connections. This setting cannot be changed for clusters in CockroachDB {{ site.data.products.cloud }}.
+[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds | How long after `server.shutdown.initial_wait` completes to wait for client connections to drain before forcibly disconnecting them from the node. Can be set to a value between 0 and 30 minutes (1800 seconds).
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:76]: I'd reword this a bit: "how long a crdb node waits for client SQL connections to close before forcibly closing the connection". 

Below, I'd also note the tradeoff involved when setting this value. Increasing this value will result in slower cluster operations (version upgrades, decommissioning nodes, etc). Each node shutdown will now wait `server.shutdown.connections.timeout` as part of the shutdown process. But increasing it may be required to avoid connection interruptions during these maintenance ops.

File: src/current/cockroachcloud/production-checklist.md [hunk 8]
```diff
@@ -66,6 +66,20 @@ When several transactions try to modify the same underlying data concurrently, t
...
+[`server.shutdown.transactions.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-query-wait)<br />Alias: `server.shutdown.query_wait` | 90 seconds | How long to wait for transactions on the node to complete before rolling them back. The default value is appropriate for most deployments. If this is set too low, transactions will be rolled back unnecessarily during node shutdown, and if it is set too high, node shutdown may take longer.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:77]: We should note that 90s is the highest supported value. Increasing it beyond this value can result in tx failures during cluster operations.

File: src/current/cockroachcloud/production-checklist.md (hunk 3 above)
Comment (carloruiz) [reply src/current/cockroachcloud/production-checklist.md:69]: On further read, I think this section should be folded under "Keeping connections current"

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md (hunk 2 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md:108]: Thank you for fixing it!

File: src/current/cockroachcloud/production-checklist.md (hunk 3 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done, created a new Networking considerations heading and moved Keeping Connections Current (merged with the shutdown settings section) and Network Authorization into it.

File: src/current/cockroachcloud/production-checklist.md (hunk 4 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done

File: src/current/cockroachcloud/production-checklist.md (hunk 6 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done

File: src/current/cockroachcloud/production-checklist.md (hunk 7 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done

File: src/current/cockroachcloud/production-checklist.md [hunk 9]
```diff
@@ -58,26 +70,28 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
+After an application establishes a connection to CockroachDB {{ site.data.products.cloud }}, the connection may become invalid. This could be due to a variety of factors, such as a change in the cluster topography, a rolling [upgrade]({% link cockroachcloud/upgrade-policy.md %}), cluster or hardware maintenance, network disruption, or cloud infrastructure unavailability.
 
 Set the maximum lifetime of a connection to between 5 and 30 minutes. {{ site.data.products.dedicated }} and {{ site.data.products.serverless }} support 30 minutes as the maximum connection lifetime. When a node is shut down or restarted, client connections can be reset after 30 minutes, causing a disruption to applications.
 
+If node termination takes longer than the sum of `server.shutdown.connections.timeout` and `server.shutdown.transactions.timeout`, regardless of the cluster's settings, the node is forcibly terminated and restarted.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:86]: Let's drop this line. As a developer it is easier to digest an upper bound on a single variable than an upper bound on an equation. Let's give upper bounds on each variable separately.  

Also, this line and everything below do not apply to Serverless.

File: src/current/cockroachcloud/production-checklist.md [hunk 10]
```diff
@@ -58,26 +70,28 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
+
+Clusters in CockroachDB {{ site.data.products.cloud }} have the following default values for [cluster settings]({% link {{ site.current_cloud_version }}/cluster-settings.md %}) that relate to [node shutdown]({% link {{ site.current_cloud_version }}/node-shutdown.md %}) for maintenance, upgrades, or scaling. Depending on the requirements of your applications and workloads, you may need to modify `server.shutdown.connections.timeout` or `server.shutdown.transactions.timeout`.
+
+Cluster setting | Default | Details
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:90]: Can we add a column for max value?

**Review by carloruiz — COMMENTED** [review]
Last few comments are about splitting dedicated and serverless recommendations. For serverless it is simpler, we just tell them to configure their connection lifetimes to a max of 30m. The cluster setting considerations only apply to dedicated.

File: src/current/cockroachcloud/production-checklist.md (hunk 10 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done. If I understand it correctly, we default to the max for `server.shutdown.transactions.timeout` so I changed the last sentence to "Decreasing this value ...."

File: src/current/cockroachcloud/production-checklist.md [hunk 11]
```diff
@@ -70,27 +77,22 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
+Cluster setting | Default | Maximum | Details
+----------------|---------|---------|-------------
+[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds  | 3 minutes (1800 seconds) | How long to wait for client connections to drain before forcibly disconnecting them from the node.
+[`server.shutdown.transactions.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-query-wait)<br />Alias: `server.shutdown.query_wait`          | 90 seconds | 90 seconds               | The longest expected duration for an incomplete transaction to complete if it is still running after `server.shutdown.connections.timeout` elapses during node shutdown. After this interval, the transaction will be rolled back to allow the node to restart. A higher value will result in slower cluster operations such as upgrades and scaling events. Decreasing this value reduces node shutdown time at the expense of running transactions during maintenance events.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:95]: `s/rolled back/cancelled`. 

Add: this value should be set to the longest expected transaction. Transactions lasting longer than `server.shutdown.transactions.timeout` can be interrupted during node restarts.

File: src/current/cockroachcloud/production-checklist.md [hunk 12]
```diff
@@ -70,27 +77,22 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
-[`server.shutdown.transactions.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-query-wait)<br />Alias: `server.shutdown.query_wait` | 90 seconds | The longest expected duration for an incomplete transaction to complete if it is still running after `server.shutdown.connections.timeout` elapses during node shutdown. After this interval, the transaction will be rolled back to allow the node to restart. A higher value will result in slower cluster operations such as upgrades and scaling events. Consider increasing this value if you experience connection disruptions during these maintenance events. Can be set to a maximum of 90 seconds.
+Cluster setting | Default | Maximum | Details
+----------------|---------|---------|-------------
+[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds  | 3 minutes (1800 seconds) | How long to wait for client connections to drain before forcibly disconnecting them from the node.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:94]: Add: connections with lifetimes over 30m can be interrupted during node restarts.

File: src/current/cockroachcloud/production-checklist.md [hunk 13]
```diff
@@ -70,27 +77,22 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
-
-For guidance on sizing, validating, and using connection pools with CockroachDB, refer to [Use Connection Pools](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/connection-pooling).
-
 <a id="keeping-connections-current"></a>
 ### Keep connections current
 
 After an application establishes a connection to CockroachDB {{ site.data.products.cloud }}, the connection may become invalid. This could be due to a variety of factors, such as a change in the cluster topography, a rolling [upgrade]({% link cockroachcloud/upgrade-policy.md %}), cluster or hardware maintenance, network disruption, or cloud infrastructure unavailability.
 
-Set the maximum lifetime of a connection to between 5 and 30 minutes. {{ site.data.products.dedicated }} and {{ site.data.products.serverless }} support 30 minutes as the maximum connection lifetime. When a node is shut down or restarted, client connections can be reset after 30 minutes, causing a disruption to applications.
+#### CockroachDB {{ site.data.products.serverless }}
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:85]: Add: set the connection lifetime in the connection pool to 30m.

File: src/current/cockroachcloud/production-checklist.md [hunk 14]
```diff
@@ -70,27 +77,22 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
 
-### Use a pool of persistent connections
-
-Creating the appropriate size pool of connections is critical to gaining maximum performance in an application. Too few connections in the pool will result in high latency as each operation waits for a connection to open up. But adding too many connections to the pool can also result in high latency as each connection thread is being run in parallel by the system. The time it takes for many threads to complete in parallel is typically higher than the time it takes a smaller number of threads to run sequentially.
-
-For guidance on sizing, validating, and using connection pools with CockroachDB, refer to [Use Connection Pools](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/connection-pooling).
-
 <a id="keeping-connections-current"></a>
 ### Keep connections current
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:81]: Let's merge Keeping Connections Current and SQL Connection Handling.

File: src/current/cockroachcloud/production-checklist.md [hunk 15]
```diff
@@ -70,27 +77,22 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
 
-Set the maximum lifetime of a connection to between 5 and 30 minutes. {{ site.data.products.dedicated }} and {{ site.data.products.serverless }} support 30 minutes as the maximum connection lifetime. When a node is shut down or restarted, client connections can be reset after 30 minutes, causing a disruption to applications.
+#### CockroachDB {{ site.data.products.serverless }}
+#### CockroachDB {{ site.data.products.dedicated }}
 
-If node termination takes longer than the sum of `server.shutdown.connections.timeout` and `server.shutdown.transactions.timeout`, regardless of the cluster's settings, the node is forcibly terminated and restarted.
+In your application server, set the maximum lifetime of a connection to between 5 and 30 minutes. When a node is shut down or restarted, client connections can be reset after 30 minutes, causing a disruption to applications.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:88]: After this line `In your application server, set the maximum lifetime of a connection to between 5 and 30 minutes.` let's also tell them to set `server.shutdown.connections.timeout` equal to the max lifetime on their connections.

**Review by carloruiz — COMMENTED** [review]
Let's have a quick call tomorrow to hammer this out with faster iteration cycles.

File: src/current/cockroachcloud/production-checklist.md (hunk 12 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:94]: Done with some tweaks. PTAL.

File: src/current/cockroachcloud/production-checklist.md (hunk 14 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:81]: Done with some wording adjustments.

File: src/current/cockroachcloud/production-checklist.md (hunk 15 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:88]: Done

**Comment by mdlinville** [comment]
> Let's have a quick call tomorrow to hammer this out with faster iteration cycles.

Sorry I wasn't able to do this today but if you have a lot more feedback on this round, let's do that tomorrow or Thursday.

File: src/current/cockroachcloud/production-checklist.md [hunk 16]
```diff
@@ -58,25 +70,26 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
+For guidance on sizing, validating, and using connection pools with CockroachDB, refer to the following sections and to [Use Connection Pools]({% link {{site.current_cloud_version}}/connection-pooling.md %}).
 
-For guidance on sizing, validating, and using connection pools with CockroachDB, refer to [Use Connection Pools](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/connection-pooling).
+#### CockroachDB {{ site.data.products.serverless }}
+#### CockroachDB {{ site.data.products.dedicated }}
 
-## Keeping connections current
+In your application server, set the maximum lifetime of a connection to between 5 and 30 minutes, and `server.shutdown.connections.timeout` equal to the maximum connection lifetime. When a node is shut down or restarted, client connections can be reset after 30 minutes, causing a disruption to applications.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:85]: `s/reset after 30 minutes/reset after server.shutdown.connections.timeout minutes`

File: src/current/cockroachcloud/production-checklist.md [hunk 17]
```diff
@@ -58,25 +70,26 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
 
-Set the maximum lifetime of a connection to between 5 and 30 minutes. {{ site.data.products.dedicated }} and {{ site.data.products.serverless }} support 30 minutes as the maximum connection lifetime. When a node is shut down or restarted, client connections can be reset after 30 minutes, causing a disruption to applications.
+Cluster setting | Default | Maximum | Details
+----------------|---------|---------|-------------
+[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds  | 3 minutes (1800 seconds) | How long to wait for client connections to drain before forcibly disconnecting them from the node. A connection with a lifetime that exceeds `server.shutdown.connections.timeout` may be interrupted during node restarts.
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:91]: typo, 3 minutes -> 30 minutes

File: src/current/cockroachcloud/production-checklist.md [hunk 18]
```diff
@@ -58,25 +70,26 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
 
-Creating the appropriate size pool of connections is critical to gaining maximum performance in an application. Too few connections in the pool will result in high latency as each operation waits for a connection to open up. But adding too many connections to the pool can also result in high latency as each connection thread is being run in parallel by the system. The time it takes for many threads to complete in parallel is typically higher than the time it takes a smaller number of threads to run sequentially.
+For guidance on sizing, validating, and using connection pools with CockroachDB, refer to the following sections and to [Use Connection Pools]({% link {{site.current_cloud_version}}/connection-pooling.md %}).
 
-For guidance on sizing, validating, and using connection pools with CockroachDB, refer to [Use Connection Pools](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/connection-pooling).
+#### CockroachDB {{ site.data.products.serverless }}
```
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:82]: Serverless customers should also configure their connection pools with connections with max lifetimes of 30m

File: src/current/cockroachcloud/production-checklist.md (hunk 14 above)
Comment (carloruiz) [reply src/current/cockroachcloud/production-checklist.md:74]: I'd restructure this a little bit. The top level title could be "SQL connection Handling" with a subsection on "connection pooling" since this is performance related, then a section for "keeping connections current" since this is for reliability.

**Review by carloruiz — COMMENTED** [review]
OK I think we're 95% there

File: src/current/cockroachcloud/production-checklist.md (hunk 16 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:85]: Done

File: src/current/cockroachcloud/production-checklist.md (hunk 14 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:74]: Apologies, I misinterpreted what you meant by "merge". Fixed.

File: src/current/cockroachcloud/production-checklist.md (hunk 18 above)
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:82]: Done

File: src/current/cockroachcloud/production-checklist.md [hunk 19]
```diff
@@ -58,25 +66,35 @@ Production clusters should not authorize `0.0.0.0/0`, which allows all networks.
...
+The following [cluster settings]({% link {{ site.current_cloud_version }}/cluster-settings.md %}) relate to [node shutdown]({% link {{ site.current_cloud_version }}/node-shutdown.md %}) for maintenance, upgrades, or scaling. Depending on the requirements of your applications and workloads, you may need to modify them.
 
-For guidance on sizing, validating, and using connection pools with CockroachDB, refer to [Use Connection Pools](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/connection-pooling).
+Cluster setting | Default | Maximum | Details
+----------------|---------|---------|-------------
+[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds  | 3 minutes (1800 seconds) | How long to wait for client connections to drain before forcibly disconnecting them from the node. A connection with a lifetime that exceeds `server.shutdown.connections.timeout` may be interrupted during node restarts.
```
Comment (mdlinville) [inline src/current/cockroachcloud/production-checklist.md:90]:
```suggestion
[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds  | 30 minutes (1800 seconds) | How long to wait for client connections to drain before forcibly disconnecting them from the node. A connection with a lifetime that exceeds `server.shutdown.connections.timeout` may be interrupted during node restarts.
```

**Comment by mdlinville** [comment]
I've resolved all feedback so I'll go ahead and merge, and we can address any feedback as a fast follow.
